contracts/src/v0.1/Prepayment.sol

```diff
@@ -254,25 +281,113 @@ contract Prepayment is Ownable, IPrepayment, ITypeAndVersion {
     /**
      * @inheritdoc IPrepayment
      */
-    function createAccount() external returns (uint64) {
+    function getAccountDetail(
+        uint64 accId
+    ) external view returns (uint256, uint256, uint256, uint256) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.getAccountDetail();
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function getSubscriptionPaid(uint64 accId, uint256 index) external view returns (bool) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.getSubscriptionPaid(index);
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function isValidReq(uint64 accId) external view returns (bool) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.isValidReq();
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function getFeeRatio(uint64 accId) external view returns (uint256) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.getFeeRatio();
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function createAccount(IAccount.AccountType accType) external returns (uint64) {
         uint64 currentAccId = sCurrentAccId + 1;
         sCurrentAccId = currentAccId;
 
-        Account acc = new Account(currentAccId, msg.sender);
+        Account acc = new Account(currentAccId, msg.sender, accType);
         sAccIdToAccount[currentAccId] = acc;
 
-        emit AccountCreated(currentAccId, address(acc), msg.sender);
+        emit AccountCreated(currentAccId, address(acc), msg.sender, accType);
         return currentAccId;
     }
 
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function updateAccountDetail(
+        uint64 accId,
+        uint256 startTime,
+        uint256 period,
+        uint256 periodReqCount,
+        uint256 subscriptionPrice
+    ) external onlyOwner {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        account.updateAccountDetail(startTime, period, periodReqCount, subscriptionPrice);
+        emit AccountDetailUpdated(accId, startTime, period, periodReqCount);
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function updateSubscriptionPaid(
+        uint64 accId,
+        uint256 index,
+        bool value
```

Comment:
Do we need to pass the `value` argument? According to the current implementation, we only pass `true` and never `false`. Do you plan to add a scenario when we want to pass a `false`?